packages/api/src/router/widgets/dns-hole.ts

```diff
@@ -0,0 +1,39 @@
+import { TRPCError } from "@trpc/server";
+
+import { PiHoleIntegration } from "@homarr/integrations";
+import type { DnsHoleSummary } from "@homarr/integrations/types";
+import { logger } from "@homarr/log";
+import { createCacheChannel } from "@homarr/redis";
+
+import { createOneIntegrationMiddleware } from "../../middlewares/integration";
+import { createTRPCRouter, publicProcedure } from "../../trpc";
+
+export const dnsHoleRouter = createTRPCRouter({
+  summary: publicProcedure
+    .unstable_concat(createOneIntegrationMiddleware("piHole"))
+    .query(async ({ ctx }) => {
+      const cache = createCacheChannel<DnsHoleSummary>(
+        `dns-hole-summary:${ctx.integration.id}`,
+      );
+
+      const { data } = await cache.consumeAsync(async () => {
+        const client = new PiHoleIntegration(
+          ctx.integration.url,
+          ctx.integration.decryptedSecrets,
+        );
+
+        return await client.getSummaryAsync().catch((err) => {
+          logger.error(err);
```

Comment:
Instead of just logging, add some details. We do not log the class / file where the issue happend so it would be useful to get more information